### PR TITLE
completeweek1assesment

### PR DESCRIPTION
**Pull request by ryanizumi16**
i was having trouble figuring out why i kept on getting undefined for the last 2 coding challenge questions

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -11,7 +11,18 @@ var temp1 = 35
...
+const checkTemp = (temp) => {
+  if(temp1 <= 212){
+    return `${temp1} is below boiling point`
+  }
+  else if(temp2 >= 212){
+    return `${temp2} is above boiling point`
+  }
+  else if(temp3 === 212){
+    return `${temp3} is at boiling point`
+  }
```
Comment (andeeisaacs) [inline code-challenges.js:23]: Just about there! Kind of does the job- your argument isnt actually being check against. You should be checking against `temp` but instead, you're hard coding the variables into the function. Additionally, if we look at how we are checking less than or greater than, you're using 'or equal to' which means, if you input 212, into the argument and the rest of the function was correct it would hit the first if statement and return "212 is below boiling point.".

File: code-challenges.js [hunk 2]
```diff
@@ -11,7 +11,18 @@ var temp1 = 35
...
+}
+console.log(checkTemp(temp1))
```
Comment (andeeisaacs) [inline code-challenges.js:25]: You're only checking a single variable into your console logs. It is SUPER important to be checking all of the given variables in your calls.

File: code-challenges.js [hunk 3]
```diff
@@ -20,6 +31,9 @@ var temp3 = 212
 
 var myNumbers1 = [3, 7, 0, 6, -9]
 
+  for(let i= 0; i < myNumbers1.length; i++) {
+    console.log(myNumbers1[i] * 5)
+  }
```
Comment (andeeisaacs) [inline code-challenges.js:36]: I'd double check the requirements on this challenge, it's asking for the for loop to be inside of a function. As it stands right now, the for-loop is just kind of floating out in the ether outside of a function.

File: code-challenges.js [hunk 4]
```diff
@@ -29,8 +43,10 @@ var myNumbers1 = [3, 7, 0, 6, -9]
 
 var myNumbers2 = [8, -7, 0, 6, 2]
 
-
-
+const mult5 = (array) => {
+  return array.map(value => value * 5)
+}
+console.log(mult5(myNumbers2))
```
Comment (andeeisaacs) [inline code-challenges.js:49]: This looks great!

File: code-challenges.js [hunk 5]
```diff
@@ -39,28 +55,60 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
-
+const noVowels = (string) => {
+  let newArray = string.split("")
+  let filteredArray = newArray.filter(value => {
+    return value !== "a" && value !== "e" && value !== "i" && value !== "o" && value !== "u"
+  })
+  let newString = filteredArray.join("")
+  return newString
+}
+console.log(noVowels(stringWithVowels2))
```
Comment (andeeisaacs) [inline code-challenges.js:66]: Almost there!!! This would output `ILvJvScrpt`, however if we look at the EXPECTED output is, we should be receiving `LvJvScrpt`.

This means, its catching a capital vowel, the i specifically. This can be fixed if we look into our filter and the conditions we're checking. I'd look into .toLowerCase() and applying it to each value check.

File: code-challenges.js [hunk 6]
```diff
@@ -39,28 +55,60 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
+//   // })
+//   // let newString = filteredArray.join("")
+//   // return newString
+//   if(true !== string){
+//     return "true is not a string"
+//   }
+//   else if(42 !== string) {
+//     return "42 is not a string"
+//   }
+// }
```
Comment (andeeisaacs) [inline code-challenges.js:88]: I think the problem here is that your return is ending your function early on, it needs to be the VERY last thing to happen in your function.

File: code-challenges.js [hunk 7]
```diff
@@ -39,28 +55,60 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
 
-
-
-
+var cats = toonimals.filter(value => {
+  if(value === "cat"){
+    return value.name
+  }
+})
+console.log(cats.name)
```
Comment (andeeisaacs) [inline code-challenges.js:102]: I'm not sure what is happening here, but I can see that this is NOT a function. Be sure to read through the instructions thoroughly.

File: code-challenges.js [hunk 8]
```diff
@@ -39,28 +55,60 @@ var myNumbers2 = [8, -7, 0, 6, 2]
...
 // Expected output: "Itchy" "Daffy" "Ren"
+
+var toonimals = [ { name: "Itchy", animal: "mouse" }, { name: "Stimpy", animal: "cat" }, { name: "Daffy", animal: "duck" }, { name: "Scratchy", animal: "cat" }, { name: "Ren", animal: "dog" }, { name: "Felix", animal: "cat" }]
+
+var nonCats = toonimals.filter(value => {
+  if(value.name !== "cat"){
+    return value.name
+  }
+})
+console.log(nonCats.name)
```
Comment (andeeisaacs) [inline code-challenges.js:114]: This is not a function, unfortunately.

File: interview-questions.md [hunk 9]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is a function? Why would you use one?
 
   Your answer:
-
+    It is a block of code that performs a specific task
```
Comment (andeeisaacs) [inline interview-questions.md:11]: I would elaborate a bit deeper here- The interviewer is also asking, why would you use one?

File: interview-questions.md [hunk 10]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
   Researched answer:
-
+    The function is invoked only when you call the function.
 
 
 2. What is the difference between .map() and .filter()?
 
   Your answer:
+    .map() is a higher order function that returns a new array where you have applied a function on the element and changes the original element.  And .filter() also returns a new array of the elements but does not change the original values.
```
Comment (andeeisaacs) [inline interview-questions.md:19]: Kind of- Both do not alter the original array. When you say 'apply a function on an element' what do you mean by function?

File: interview-questions.md [hunk 11]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Researched answer:
-
+    .filter() returns subset of the elements from the original array. and .map() produces and array with a new and different entries based on the elements in the original array.
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+    console.log() strictly is for your terminal that logs whatever arguments you call.  and return is a statement that gives an output from a functions
```
Comment (andeeisaacs) [inline interview-questions.md:28]: Totally!

File: interview-questions.md [hunk 12]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+    .filter() returns subset of the elements from the original array. and .map() produces and array with a new and different entries based on the elements in the original array.
 
 
 3. What is the difference between console.log() and return?
 
   Your answer:
+    console.log() strictly is for your terminal that logs whatever arguments you call.  and return is a statement that gives an output from a functions     
 
   Researched answer:
+    console.log() has no practical use to a program.  return allows you to pass the variable's value to the calling scope.
```
Comment (andeeisaacs) [inline interview-questions.md:31]: Not quite, I'd definitely take some time and look into Return a bit more.

File: interview-questions.md [hunk 13]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 4. In regards to functions, what is an argument?
 
   Your answer:
+    its an accessible object inside a function that has values passed into a function.
```
Comment (andeeisaacs) [inline interview-questions.md:38]: Not exactly, an argument object and an object, are very specific things.  An object can be passed in AS an argument, but it is not specifically an argument as the interviewer is asking.  I think what you might be looking for is parameter.

File: interview-questions.md [hunk 14]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Researched answer:
-
+    All arguments passed to a JavaScript function can be referred using the arguments object.
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
   Your answer:
+      steady communication, making sure each person takes equal ammount of turns navigating and driving.  Being respectul and patient.  The roles of each person in paired programming is one person is called the "driver" where that person is sharing their screen is the one that is actually doing the coding, and the other person is called the "navigator" where that person is looking ahead and figuring out problems and double checking the drivers work
```
Comment (andeeisaacs) [inline interview-questions.md:47]: Great!  Id also mention how the roles can be switched after a given period of time.

File: interview-questions.md [hunk 15]
```diff
@@ -8,63 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
   Researched answer:
-
+    paired proggraming is used for higher quality code, it is faster and it also reduces risk.
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
   Your answer:
+      TDD stands for test driven development, tests are developed to specify and validate what the code will do.  you first have to write code that fails in order for you to write new functional code.
```
Comment (andeeisaacs) [inline interview-questions.md:56]: I would maybe mention a bit more about 'red-green-refactor', and specify what you mean by 'write code that fails'

File: testing-with-jest.test.js [hunk 16]
```diff
@@ -8,20 +8,38 @@
 
 // --------------------1a) Create a test for a function called addThemUp that takes two numbers as arguments and returns the sum.
 
+describe("addThemUp", () => {
+  test("function called addThemUp that takes two numbers as arguments and returns the sum", () => {
+    expect(addThemUp(2, 5)).toEqual(7)
+  })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:15]: Great! I'd want to encourage you to write additional expects to check for more edge cases! What if the function was passed something that was not integer datatypes?

File: testing-with-jest.test.js [hunk 17]
```diff
@@ -8,20 +8,38 @@
...
 
 
 // --------------------2a) Create a test for a function called triangleArea that takes the base and the height of a triange and returns the area. The area of a triange is the base times the height divided by two.
 
-
+describe("triangleArea", () => {
+  test("function called triangleArea that takes the base and the height of a triange and returns the area. The area of a triange is the base times the height divided by two", () => {
+    expect(triangleArea(6, 6)).toEqual(18)
+  })
+})
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:34]: Excellent!

File: testing-with-jest.test.js [hunk 18]
```diff
@@ -8,20 +8,38 @@
...
 
 
 // --------------------2b) See the test fail. THEN write the code to make the test pass.
+
+const triangleArea = (num1, num2) => {
+return 6 * 6 / 2
+
+
+
+}
```
Comment (andeeisaacs) [inline testing-with-jest.test.js:45]: Your spacing and indentation is a bit off here. Besides that, you're also hard-coding the return results, thus negating the purpose of the function. What if 12 and 12 was passed in to the function call? The arguments of the function are essentially useless at this point. 

Additionally, the arguments that should be passed in come from the expect, it's your function call.

**Review by andeeisaacs — APPROVED** [review]
Ryan- 

There is a lot of room for improvement here. I look forward to seeing how you do with the week 2 assessment. 

Some pieces of advice- 

- I noticed you didn't pseudo code any of your work. Pseudo coding is SUPER important as its a way we communicate what our code is doing below it. It helps not only yourself plan out what you are going to do, or your future self understand what you were doing, but also anyone else who is looking at your code. Companies will require you to perform pull requests, just like this, where a senior developer will be reviewing your code- and they will need to be able to understand your thought process and flow to find solutions to the problems are you faced with. Pseudo coding does this for you! I encourage you to pseudo code EVERYTHING you possibly can, to the point it may even come across annoying! :)

- I'd encourage you to gain a deeper understanding of the purpose of a function. Many of your functions were hard-coded and would not work if anything other than the given variables/inputs were passed into them. One of the best places you can do this is to review the first week units in the syllabus. 

- While your tests were pretty good, the code that allows them to pass were not. Much like your JS challenges functions, your test functions were hard coded. I would encourage you to take a deep dive into understanding testing and the core concepts of functions. 


If you have any questions, please don't hesitate to ask! Feel free to merge to master.
